substrate/utils/frame/remote-externalities/src/lib.rs

```diff
@@ -523,20 +525,39 @@ where
 
 				log::debug!(
 					target: LOG_TARGET,
-					"Batch request failed, trying again with smaller batch size. {}",
+					"Batch request failed, resetting batch size to 1. Error: {}",
 					e.to_string()
 				);
 
-				return Self::get_storage_data_dynamic_batch_size(
-					client,
-					payloads,
-					max(1, (batch_size as f32 * Self::BATCH_SIZE_DECREASE_FACTOR) as usize),
-					bar,
-				)
-				.await
+				// Request timed out or server errored. This is very bad. Try to get things moving
+				// again by starting again with just 1 item.
+				return Self::get_storage_data_dynamic_batch_size(client, payloads, 1, bar).await
```

Comment:
If the connection is closed or something, we will run here into a out of stack error.